include/fixed_containers/wyhash.hpp

```diff
@@ -0,0 +1,204 @@
+#pragma once
+
+#if defined(__clang__) && __clang_major__ >= 15
+#include "fixed_containers/reflection.hpp"
```

Comment:
We don't want a clang-specific solution for this.

Reply:
This is the same guard that the entire reflection library uses.
It isn't a "solution" so much as an "extension"... `std::hash` doesn't have anything comparable, but I figured since this library has reflection anyway then I should use it to give this extra functionality when possible.